ams/server/src/main/java/com/netease/arctic/server/table/DefaultTableService.java

```diff
@@ -358,36 +389,84 @@ public void dispose() {
                 .collect(Collectors.toMap(TableIdentity::new, tableIdentifier -> tableIdentifier));
+        LOG.info("Loaded {} tables from Amoro server catalog {}.",
+            serverTableIdentifiers.size(), externalCatalog.name());
         Sets.difference(tableIdentifiers, serverTableIdentifiers.keySet())
             .forEach(tableIdentity -> {
-              try {
-                syncTable(externalCatalog, tableIdentity);
-              } catch (Exception e) {
-                LOG.error("TableExplorer sync table {} error", tableIdentity.toString(), e);
-              }
-            });
+                  try {
+                    tableExplorerExecutors.submit(
+                        () -> {
+                          try {
+                            syncTable(externalCatalog, tableIdentity);
+                          } catch (Exception e) {
+                            if (e.getCause() != null &&
+                                e.getCause() instanceof NoSuchIcebergTableException &&
+                                e.getMessage().contains("Not an iceberg table")) {
+                              LOG.info("Skip non-iceberg table {}.", tableIdentity.toString());
+                            } else {
+                              LOG.error("TableExplorer sync table {} error", tableIdentity.toString(), e);
+                            }
+                          }
+                        });
+                  } catch (RejectedExecutionException e) {
+                    LOG.error("The queue of table explorer is full, please increase the queue size or thread count.");
+                  }
+                }
+            );
         Sets.difference(serverTableIdentifiers.keySet(), tableIdentifiers)
             .forEach(tableIdentity -> {
               try {
-                disposeTable(externalCatalog, serverTableIdentifiers.get(tableIdentity));
-              } catch (Exception e) {
-                LOG.error("TableExplorer dispose table {} error", tableIdentity.toString(), e);
+                tableExplorerExecutors.submit(
+                    () -> {
+                      try {
+                        disposeTable(externalCatalog, serverTableIdentifiers.get(tableIdentity));
+                      } catch (Exception e) {
+                        LOG.error("TableExplorer dispose table {} error", tableIdentity.toString(), e);
+                      }
+                    });
+              } catch (RejectedExecutionException e) {
+                LOG.error("The queue of table explorer is full, please increase the queue size or thread count.");
               }
             });
-      } catch (Exception e) {
-        LOG.error("TableExplorer run error", e);
+      } catch (Throwable e) {
+        LOG.error("TableExplorer error", e);
       }
     }
+    long end = System.currentTimeMillis();
+    LOG.info("Syncing external catalogs took {} ms.", end - start);
```

Comment:
Should we wait for all tasks to complete to avoid concurrent execution of the refresh process?

BTW, some unit test cases depend on the synchronization of tables being completed after the `exploreExternalCatalog` completed. The current implementation cannot guarantee that the synchronization will be completed after the call is finished, which is the reason for some unit test failures.

Reply:
`org.apache.iceberg.util.Taks` is a useful tool to help you execute tasks in parallel more conveniently. I hope this is helpful for you.